### PR TITLE
create a dedicated testing project

### PR DESCRIPTION
**Pull request by dawedawe**
To have cleaner dependencies, add a dedicated project/package to help analyzer authors with testing.

I'm not sure what to do with the `Authors` tag in the `.fsproj` file. I don't want to list people without their consent, but I also don't want to exclude people. So please just tell me, if you want to be listed as an author.

### CODE REVIEW

**Review by TheAngryByrd — APPROVED** [review]
❤️ 

Re: Authors, you did the work you get the credit.

Might wanna consider moving all that common packaging stuff to a `Directory.build.props`

**Comment by nojaf** [comment]
> Might wanna consider moving all that common packaging stuff to a Directory.build.props

We do similar stuff in

https://github.com/fsprojects/fantomas/blob/db9e6e4a451767b4e09c33b9a4fbbb1a70a812e1/Directory.Build.props#L55-L60

File: src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj [hunk 1]
```diff
@@ -7,26 +7,19 @@
     <Authors>Krzysztof Cieslak</Authors>
     <Description>SDK for building custom analyzers for FSAC / F# editors</Description>
     <Copyright>Copyright 2019 Lambda Factory</Copyright>
-    <PackageLicenseExpression>MIT</PackageLicenseExpression>
```
Comment (nojaf) [inline src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj:10]: You will need to set `<IsPackable>true</IsPackable>` in order for that information to be picked up by the Directory.Props file

Comment (dawedawe) [reply src/FSharp.Analyzers.SDK/FSharp.Analyzers.SDK.fsproj:10]: Mmh, I see the same metadata with or without `IsPackable` 🤷

**Review by nojaf — APPROVED** [review]
Good stuff. @TheAngryByrd could we maybe have another release near the end of the week?

**Comment by TheAngryByrd** [comment]
> Good stuff. @TheAngryByrd could we maybe have another release near the end of the week?

If you can push to master you should be able to [release](https://github.com/ionide/FSharp.Analyzers.SDK/blob/master/CONTRIBUTING.md#release)

**Pull request merged**